### PR TITLE
Add a MediaDevices proxy to support the "devicechange" in all supported browsers

### PR DESCRIPTION
**Pull request by simmkyu**
**Issue #:** 
https://github.com/aws/amazon-chime-sdk-js/issues/445

**Description of changes:**
- Add a factory that creates the MediaDevices proxy. In a browser without the `devicechange` support, the proxy will poll a list of devices every one second and call listeners if the list changes. It will start polling only if there are one or more listeners.
- Why not using [the existing logic `DefaultDeviceController.handleDeviceChange()`](https://github.com/aws/amazon-chime-sdk-js/blob/8e5802c04031ac62a2c5c5fe75af5bb4ffbf12d0/src/devicecontroller/DefaultDeviceController.ts#L63)? (I could make it work by adding more conditions, but I prefer not to modify it.)
  - `handleDeviceChange()` calls `getUserMedia()` when a label doesn't exist.
  - `handleDeviceChange()` calls itself recursively to defer a task.
  - `handleDeviceChange()` uses all device info fields for comparison.

**Testing**

1. Have you successfully run `npm run build:release` locally?
  Yes
2. How did you test these changes?
  Tested by connecting and disconnecting devices.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by simmkyu** [comment]
Thanks Trung and Andrew for reviewing. We decided to polyfill `MediaDevices.addEventListener` instead of adding a new component.

File: src/devicecontroller/DefaultDeviceController.ts [hunk 1]
```diff
@@ -456,6 +462,7 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
   private forEachObserver(observerFunc: (observer: DeviceChangeObserver) => void): void {
     for (const observer of this.deviceChangeObservers) {
       new AsyncScheduler().start(() => {
+        /* istanbul ignore else */
```
Comment (XHatan) [inline src/devicecontroller/DefaultDeviceController.ts:465]: what is "istanbul ignore else " ?

Comment (simmkyu) [reply src/devicecontroller/DefaultDeviceController.ts:465]: Istanbul will skip the `else` case for coverage. Unless we make `forEachObserver` public, it's impossible to cover the `else` case.

SDK is iterating listeners in `this.deviceChangeObservers`. In the next tick (`setTimeout` with zero delay), SDK checks whether the listener exists in `this.deviceChangeObservers`.

**Comment by simmkyu** [comment]
Rebased

**Pull request merged**